pandas/io/stata.py

```diff
@@ -1345,6 +1358,307 @@ def data(self, convert_dates=True, convert_categoricals=True, index=None,
 
         return data
 
+
+    def get_chunk(self, chunksize, convert_dates=True,
+             convert_categoricals=True, convert_missing=False,
+             preserve_dtypes=True, columns=None,
+             order_categoricals=True, skip_data=False):
+        """
+        Reads observations from Stata file, converting them into a dataframe
+
+        Parameters
+        ----------
+        chunksize : int
+            Number of lines to read from data file
+        convert_dates : boolean, defaults to True
+            Convert date variables to DataFrame time values
+        convert_categoricals : boolean, defaults to True
+            Read value labels and convert columns to Categorical/Factor
+            variables
+        convert_missing : boolean, defaults to False
+            Flag indicating whether to convert missing values to their Stata
+            representation.  If False, missing values are replaced with
+            nans.  If True, columns containing missing values are returned with
+            object data types and missing values are represented by
+            StataMissingValue objects.
+        preserve_dtypes : boolean, defaults to True
+            Preserve Stata datatypes. If False, numeric data are upcast to
+            pandas default types for foreign data (float64 or int64)
+        columns : list or None
+            Columns to retain.  Columns will be returned in the given order.
+            None returns all columns
+        order_categoricals : boolean, defaults to True
+            Flag indicating whether converted categorical data are ordered.
+        skip_data : boolean
+            Don't process any data and return an empty DataFrame.
+            Used to advance quickly to the label information in pre
+            version 117 files.
+
+        Returns
+        -------
+        y : DataFrame instance
+
+        Notes
+        -----
+        Some dtypes may differ when reading by chunks and when reading
+        the entire file with the `data` method.  For example, if there
+        are missing values in an integer dtype column, the column will
+        be upcast to float only if there are missing values in a
+        particular chunk.
+
+        For file format versions below 117, `convert_categoricals`
+        must be set to False.  After the last chunk has been read, the
+        category label information will be available in the iterator's
+        `lbllist` and `value_label_dict` fields.
+        """
+        self._missing_values = convert_missing
+
+        if (self.format_version < 117) and convert_categoricals:
+            raise ValueError("For format versions < 117, `convert_categoricals` must be set to False.  See docstring for details.")
+
+        first_chunk = False
+        if not hasattr(self, '_not_first_chunk'):
```

Comment:
Adding attributes as a substitute for a `bool` doesnt' seem like a good idea.  Why not just add `_first_chunk=True` in `__init__`?  If not using the iterator it will just be ignored.